### DISCUSSION TITLE
Somehow ignorant(?) question about jujutsu

### DISCUSSION THREAD

**Discussion by a-man-called-q** [jj-vcs/jj#4792] [General]
Hey, I stumbled to https://www.youtube.com/watch?v=LV0JzI8IcCY when I was opening youtube. `jj` seems interesting. However, I have questions about `jj`:

- Is `jj` a new DVCS like `git` but can be used fully independent without `git` or `jj` is a git-related VCS(?) that makes `git` operation easier and must have `git` as dependency (because lots of people use `git` as the data storage for `jj`)?
- Does `jj` have some kind of `auth` implemented? If so, where can I find the docs?  or if not, does `jj` have "hooks" like [`git`](https://git-scm.com/book/en/v2/Customizing-Git-Git-Hooks)?

Sorry if this sounds too shallow, I only use `git` for daily usage and "just enough" to get by.

**Comment by joyously** [comment]
Please read the README, where it says
> Jujutsu is unlike most other systems, because internally it abstracts the user interface and version control algorithms from the storage systems used to serve your content. This allows it to serve as a VCS with many possible physical backends

and
> Today, we use Git repositories as a storage layer to serve and track content

For `auth`, do you mean commit signing? Read about that on the [Settings](https://martinvonz.github.io/jj/latest/config/#commit-signing) page.

For hooks, there are none yet, but issues are open, like https://github.com/martinvonz/jj/issues/3577.

**Comment by a-man-called-q** [comment]
@joyously thanks for the response

> Please read the README, where it says
> 
>> Jujutsu is unlike most other systems, because internally it abstracts the user interface and version control algorithms from the storage systems used to serve your content. This allows it to serve as a VCS with many possible physical backends
> 
> and
> 
> >Today, we use Git repositories as a storage layer to serve and track content

Yes I read that before I asked here, sorry for not mentioning before. It's because of ` This allows it to serve as a VCS with many possible physical backends` I became confused :smile: 

> For auth, do you mean commit signing? Read about that on the [Settings](https://martinvonz.github.io/jj/latest/config/#commit-signing) page.

No, I mean permission to access the repo. like RBAC or something simpler. 

> For hooks, there are none yet, but issues are open, like https://github.com/martinvonz/jj/issues/3577.

Thanks !

**Comment by martinvonz** [comment]
> * Is `jj` a new DVCS like `git` but can be used fully independent without `git`

Yes, it's independent of Git. We custom storage backends and commands at Google. Git is not involved there. There's a minute of two towards the end of [my demo at Git Merge 2024](https://youtu.be/LV0JzI8IcCY?si=3uhLgdp8Ls302Ym8) where I showed that it works and looks pretty much like regular jj on Git.

Well, jj is not necessarily a distributed VCS like Git. It depends on the storage backends. The storage backends at Google store all the data in a central database.

> or `jj` is a git-related VCS(?) that makes `git` operation easier and must have `git` as dependency (because lots of people use `git` as the data storage for `jj`)?

Most of jj is agnostic of the storage backend implementation. The `jj git` commands add some interop specifically for Git, however. We have `jj piper` commands at Google for interop without internal VCS (which is called Piper).

> * Does `jj` have some kind of `auth` implemented? If so, where can I find the docs?  or if not, does `jj` have "hooks" like [`git`](https://git-scm.com/book/en/v2/Customizing-Git-Git-Hooks)?

No, it depends on the forge-specific integration. For Git, we currently use libgit2 with support for https and ssh (currently libssh2). At Google, we use gRPC with Google-specific authentication.

**Comment by a-man-called-q** [comment]
> Well, jj is not necessarily a distributed VCS like Git. It depends on the storage backends. The storage backends at Google store all the data in a central database.

Can `jj` works alone ? From your statement it seems like `jj` must have somekind of storage backend that isn't included with `jj` although I saw a command like `jj init` in the documentation.

**Reply by ilyagr** [reply]
The git backend is built-in and compiled into the `jj` binary. `jj` currently is quite self-sufficient.

We might add an option to shell out to a git binary in the future, but it'd be an option, and it doesn't exist yet.

**Reply by a-man-called-q** [reply]
Thanks for the explanations

**Comment by a-man-called-q** [comment]
@ilyagr @martinvonz @joyously I'm going to close the discussion. Thanks for the responses. I'll test drive `jj` for a while.
